### PR TITLE
feat(plasma-b2c, plasma-web): Modal refinement

### PR DESCRIPTION
**Pull request by kayman233**
Добавлены новые свойства для компонента `Modal`:
- `сloseOnEsc?: boolean` - Закрывать модальное окно при нажатии на ESC(по умолчанию true).
- `сloseOnOverlayClick?: boolean` - Закрывать модальное окно при нажатии вне области модального окна(по умолчанию true).
- `onEscKeyDown?: (event: KeyboardEvent) => void` - Обработчик клика при нажатии на ESC(если не передан, то при нажатии используется onClose).
- `onOverlayClick?: (event: React.MouseEvent<HTMLDivElement>) => void` - Обработчик клика при нажатии вне области модального окна(если не передан, то при нажатии используется onClose).
- `initialFocusRef?: React.RefObject<HTMLElement>` - Первый элемент для фокуса внутри модального окна.
- `focusLaterRef?: React.RefObject<HTMLElement>` - Элемент для фокуса после закрытия модального окна(по умолчанию фокус на последнем перед открытием активном элементе).
- `showCloseButton?: boolean` - Отображать кнопку "закрыть"(по умолчанию true).

<details>
  <summary>:baby_chick: Download canary assets:</summary>
  <blockquote>
    <details>
      <summary>color</summary>
      <blockquote>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_plasma_b2c_ios-swift--canary.629.5736515470.swift'>color_plasma_b2c_ios-swift--canary.629.5736515470.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_plasma_b2c_kotlin--canary.629.5736515470.kt'>color_plasma_b2c_kotlin--canary.629.5736515470.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_plasma_b2c_react-native--canary.629.5736515470.ts'>color_plasma_b2c_react-native--canary.629.5736515470.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_plasma_b2c_xml--canary.629.5736515470.xml'>color_plasma_b2c_xml--canary.629.5736515470.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_ios-swift--canary.629.5736515470.swift'>color_sberHealth_ios-swift--canary.629.5736515470.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_kotlin--canary.629.5736515470.kt'>color_sberHealth_kotlin--canary.629.5736515470.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_react-native--canary.629.5736515470.ts'>color_sberHealth_react-native--canary.629.5736515470.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_xml--canary.629.5736515470.xml'>color_sberHealth_xml--canary.629.5736515470.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_ios-swift--canary.629.5736515470.swift'>color_sbermarket_business_ios-swift--canary.629.5736515470.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_kotlin--canary.629.5736515470.kt'>color_sbermarket_business_kotlin--canary.629.5736515470.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_react-native--canary.629.5736515470.ts'>color_sbermarket_business_react-native--canary.629.5736515470.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_xml--canary.629.5736515470.xml'>color_sbermarket_business_xml--canary.629.5736515470.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.629.5736515470.swift'>color_sbermarket_ios-swift--canary.629.5736515470.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.629.5736515470.kt'>color_sbermarket_kotlin--canary.629.5736515470.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_ios-swift--canary.629.5736515470.swift'>color_sbermarket_metro_ios-swift--canary.629.5736515470.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_kotlin--canary.629.5736515470.kt'>color_sbermarket_metro_kotlin--canary.629.5736515470.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_react-native--canary.629.5736515470.ts'>color_sbermarket_metro_react-native--canary.629.5736515470.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_xml--canary.629.5736515470.xml'>color_sbermarket_metro_xml--canary.629.5736515470.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.629.5736515470.ts'>color_sbermarket_react-native--canary.629.5736515470.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_ios-swift--canary.629.5736515470.swift'>color_sbermarket_selgros_ios-swift--canary.629.5736515470.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_kotlin--canary.629.5736515470.kt'>color_sbermarket_selgros_kotlin--canary.629.5736515470.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_react-native--canary.629.5736515470.ts'>color_sbermarket_selgros_react-native--canary.629.5736515470.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_xml--canary.629.5736515470.xml'>color_sbermarket_selgros_xml--canary.629.5736515470.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_ios-swift--canary.629.5736515470.swift'>color_sbermarket_wlbusiness_ios-swift--canary.629.5736515470.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_kotlin--canary.629.5736515470.kt'>color_sbermarket_wlbusiness_kotlin--canary.629.5736515470.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_react-native--canary.629.5736515470.ts'>color_sbermarket_wlbusiness_react-native--canary.629.5736515470.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_xml--canary.629.5736515470.xml'>color_sbermarket_wlbusiness_xml--canary.629.5736515470.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_xml--canary.629.5736515470.xml'>color_sbermarket_xml--canary.629.5736515470.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_ios-swift--canary.629.5736515470.swift'>color_sberonline_ios-swift--canary.629.5736515470.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_kotlin--canary.629.5736515470.kt'>color_sberonline_kotlin--canary.629.5736515470.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_react-native--canary.629.5736515470.ts'>color_sberonline_react-native--canary.629.5736515470.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_xml--canary.629.5736515470.xml'>color_sberonline_xml--canary.629.5736515470.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.629.5736515470.swift'>color_sberprime_ios-swift--canary.629.5736515470.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.629.5736515470.kt'>color_sberprime_kotlin--canary.629.5736515470.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.629.5736515470.ts'>color_sberprime_react-native--canary.629.5736515470.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_xml--canary.629.5736515470.xml'>color_sberprime_xml--canary.629.5736515470.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_stylesSalute_ios-swift--canary.629.5736515470.swift'>color_stylesSalute_ios-swift--canary.629.5736515470.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_stylesSalute_kotlin--canary.629.5736515470.kt'>color_stylesSalute_kotlin--canary.629.5736515470.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_stylesSalute_react-native--canary.629.5736515470.ts'>color_stylesSalute_react-native--canary.629.5736515470.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_stylesSalute_xml--canary.629.5736515470.xml'>color_stylesSalute_xml--canary.629.5736515470.xml</a><br>
      </blockquote>
    </details>
    <details>
      <summary>shadow</summary>
      <blockquote>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_plasma_b2c_react-native--canary.629.5736515470.ts'>shadow_plasma_b2c_react-native--canary.629.5736515470.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sberHealth_react-native--canary.629.5736515470.ts'>shadow_sberHealth_react-native--canary.629.5736515470.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_business_react-native--canary.629.5736515470.ts'>shadow_sbermarket_business_react-native--canary.629.5736515470.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_metro_react-native--canary.629.5736515470.ts'>shadow_sbermarket_metro_react-native--canary.629.5736515470.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.629.5736515470.ts'>shadow_sbermarket_react-native--canary.629.5736515470.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_selgros_react-native--canary.629.5736515470.ts'>shadow_sbermarket_selgros_react-native--canary.629.5736515470.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_wlbusiness_react-native--canary.629.5736515470.ts'>shadow_sbermarket_wlbusiness_react-native--canary.629.5736515470.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sberonline_react-native--canary.629.5736515470.ts'>shadow_sberonline_react-native--canary.629.5736515470.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sberprime_react-native--canary.629.5736515470.ts'>shadow_sberprime_react-native--canary.629.5736515470.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_stylesSalute_react-native--canary.629.5736515470.ts'>shadow_stylesSalute_react-native--canary.629.5736515470.ts</a><br>
      </blockquote>
    </details>
    <details>
      <summary>typo</summary>
      <blockquote>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.629.5736515470.swift'>typo_mage_ios-swift--canary.629.5736515470.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.629.5736515470.kt'>typo_mage_kotlin--canary.629.5736515470.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.629.5736515470.ts'>typo_mage_react-native--canary.629.5736515470.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.629.5736515470.swift'>typo_plasma_ios-swift--canary.629.5736515470.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.629.5736515470.kt'>typo_plasma_kotlin--canary.629.5736515470.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.629.5736515470.ts'>typo_plasma_react-native--canary.629.5736515470.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.629.5736515470.swift'>typo_ruler_ios-swift--canary.629.5736515470.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.629.5736515470.kt'>typo_ruler_kotlin--canary.629.5736515470.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.629.5736515470.ts'>typo_ruler_react-native--canary.629.5736515470.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.629.5736515470.swift'>typo_sage_ios-swift--canary.629.5736515470.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.629.5736515470.kt'>typo_sage_kotlin--canary.629.5736515470.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.629.5736515470.ts'>typo_sage_react-native--canary.629.5736515470.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.629.5736515470.swift'>typo_sbermarket_ios-swift--canary.629.5736515470.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.629.5736515470.kt'>typo_sbermarket_kotlin--canary.629.5736515470.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.629.5736515470.ts'>typo_sbermarket_react-native--canary.629.5736515470.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.629.5736515470.swift'>typo_soulmate_ios-swift--canary.629.5736515470.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.629.5736515470.kt'>typo_soulmate_kotlin--canary.629.5736515470.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.629.5736515470.ts'>typo_soulmate_react-native--canary.629.5736515470.ts</a><br>
      </blockquote>
    </details>
  </blockquote>
</details>

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.220.0-canary.629.5736515470.0
  npm install @salutejs/plasma-core@1.125.0-canary.629.5736515470.0
  npm install @salutejs/plasma-hope@1.220.0-canary.629.5736515470.0
  npm install @salutejs/plasma-icons@1.154.0-canary.629.5736515470.0
  npm install @salutejs/plasma-temple@1.169.0-canary.629.5736515470.0
  npm install @salutejs/plasma-tokens-b2b@1.24.0-canary.629.5736515470.0
  npm install @salutejs/plasma-tokens-b2c@0.33.0-canary.629.5736515470.0
  npm install @salutejs/plasma-tokens-web@1.39.0-canary.629.5736515470.0
  npm install @salutejs/plasma-tokens@1.58.0-canary.629.5736515470.0
  npm install @salutejs/plasma-typo@0.36.0-canary.629.5736515470.0
  npm install @salutejs/plasma-ui@1.201.0-canary.629.5736515470.0
  npm install @salutejs/plasma-web@1.220.0-canary.629.5736515470.0
  npm install @salutejs/plasma-cy-utils@0.65.0-canary.629.5736515470.0
  npm install @salutejs/plasma-sb-utils@0.123.0-canary.629.5736515470.0
  npm install @salutejs/plasma-tokens-utils@0.29.0-canary.629.5736515470.0
  # or 
  yarn add @salutejs/plasma-b2c@1.220.0-canary.629.5736515470.0
  yarn add @salutejs/plasma-core@1.125.0-canary.629.5736515470.0
  yarn add @salutejs/plasma-hope@1.220.0-canary.629.5736515470.0
  yarn add @salutejs/plasma-icons@1.154.0-canary.629.5736515470.0
  yarn add @salutejs/plasma-temple@1.169.0-canary.629.5736515470.0
  yarn add @salutejs/plasma-tokens-b2b@1.24.0-canary.629.5736515470.0
  yarn add @salutejs/plasma-tokens-b2c@0.33.0-canary.629.5736515470.0
  yarn add @salutejs/plasma-tokens-web@1.39.0-canary.629.5736515470.0
  yarn add @salutejs/plasma-tokens@1.58.0-canary.629.5736515470.0
  yarn add @salutejs/plasma-typo@0.36.0-canary.629.5736515470.0
  yarn add @salutejs/plasma-ui@1.201.0-canary.629.5736515470.0
  yarn add @salutejs/plasma-web@1.220.0-canary.629.5736515470.0
  yarn add @salutejs/plasma-cy-utils@0.65.0-canary.629.5736515470.0
  yarn add @salutejs/plasma-sb-utils@0.123.0-canary.629.5736515470.0
  yarn add @salutejs/plasma-tokens-utils@0.29.0-canary.629.5736515470.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-629/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!

  website: http://plasma.sberdevices.ru/pr/pr-629/
b2c storybook: http://plasma.sberdevices.ru/pr/pr-629/b2c-storybook/
web storybook: http://plasma.sberdevices.ru/pr/pr-629/web-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-629/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!

  website: http://plasma.sberdevices.ru/pr/pr-629/
b2c storybook: http://plasma.sberdevices.ru/pr/pr-629/b2c-storybook/
web storybook: http://plasma.sberdevices.ru/pr/pr-629/web-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

File: packages/plasma-hope/src/hooks/useFocusTrap.ts [hunk 1]
```diff
@@ -5,11 +5,17 @@ import { focusSelector, isFocusable, isTabble } from '../utils/tabbable';
...
+    node: HTMLElement,
+    firstFocusSelector?: string | React.RefObject<HTMLElement>,
+): HTMLElement | null => {
     let focusElement: HTMLElement | null = null;
     if (firstFocusSelector) {
-        focusElement =
-            typeof firstFocusSelector === 'string' ? node.querySelector(firstFocusSelector) : firstFocusSelector;
+        if (typeof firstFocusSelector === 'string') {
+            focusElement = node.querySelector(firstFocusSelector);
+        } else if (firstFocusSelector && firstFocusSelector.current) {
```
Comment (Yakutoc) [inline packages/plasma-hope/src/hooks/useFocusTrap.ts:16]: @kayman233 Что-то я тут сломался )

Почему снова проверяем на `firstFocusSelector`?

File: packages/plasma-hope/src/utils/focusManager.ts [hunk 2]
```diff
@@ -25,8 +25,12 @@ export class FocusManager {
     };
 
     // добавление на фокус после анмаунта
-    public markForFocusLater = () => {
-        this.focusLaterElements.push(document.activeElement as HTMLElement);
+    public markForFocusLater = (focusLaterNode?: React.RefObject<HTMLElement>) => {
+        if (focusLaterNode && focusLaterNode.current) {
```
Comment (Yakutoc) [inline packages/plasma-hope/src/utils/focusManager.ts:29]: [Вкусовщина]

А если сделать через тернарный оператор?

```js

const node  =  focusLaterNode && focusLaterNode.current ? focusLaterNode.current :  document.activeElement as HTMLElement;

this.focusLaterElements.push(node);

```

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-629/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!

  website: http://plasma.sberdevices.ru/pr/pr-629/
b2c storybook: http://plasma.sberdevices.ru/pr/pr-629/b2c-storybook/
web storybook: http://plasma.sberdevices.ru/pr/pr-629/web-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Review by Yakutoc — APPROVED** [review]

File: packages/plasma-hope/src/components/Modal/Modal.tsx [hunk 3]
```diff
@@ -116,13 +141,26 @@ const NoScroll = createGlobalStyle`
...
+    onOverlayClick,
+    onEscKeyDown,
+    isCloseOnEsc = true,
+    isCloseOnOverlayClick = true,
+    withBlur,
+    initialFocusRef,
+    focusLaterRef,
+    ...rest
+}) => {
+    console.log('isCloseOnEsc', isCloseOnEsc);
```
Comment (neretin-trike) [inline packages/plasma-hope/src/components/Modal/Modal.tsx:157]: кажется лишний

File: packages/plasma-hope/src/components/Modal/Modal.tsx [hunk 4]
```diff
@@ -181,6 +225,19 @@ export const Modal: FC<ModalProps> = ({ id, isOpen, onClose, withBlur, ...rest }
         return null;
     }
 
+    const onOverlayKeyDown = (event: React.MouseEvent<HTMLDivElement>) => {
```
Comment (neretin-trike) [inline packages/plasma-hope/src/components/Modal/Modal.tsx:228]: оберни пожалуйста в `useCallBack`

File: packages/plasma-hope/src/components/Modal/ModalView.tsx [hunk 5]
```diff
@@ -11,13 +11,17 @@ export interface ModalViewProps extends React.HTMLAttributes<HTMLDivElement> {
...
      */
     onClose?: () => void;
     /**
      * WAI-ARIA атрибут кнопки "закрыть".
      */
     closeButtonAriaLabel?: string;
+    /**
+     * Отображать кнопку "закрыть"(по умолчанию true).
+     */
+    showCloseIcon?: boolean;
```
Comment (neretin-trike) [inline packages/plasma-hope/src/components/Modal/ModalView.tsx:24]: я бы на этот кейс добавил небольшой тест

File: packages/plasma-hope/src/hooks/useFocusTrap.ts [hunk 6]
```diff
@@ -5,11 +5,17 @@ import { focusSelector, isFocusable, isTabble } from '../utils/tabbable';
...
     let focusElement: HTMLElement | null = null;
     if (firstFocusSelector) {
-        focusElement =
-            typeof firstFocusSelector === 'string' ? node.querySelector(firstFocusSelector) : firstFocusSelector;
+        if (typeof firstFocusSelector === 'string') {
+            focusElement = node.querySelector(firstFocusSelector);
+        } else if (firstFocusSelector.current) {
+            focusElement = firstFocusSelector.current;
+        }
     }
```
Comment (neretin-trike) [inline packages/plasma-hope/src/hooks/useFocusTrap.ts:19]: давай плоские ифы сделаем? 

```ts
if (firstFocusSelector && typeof firstFocusSelector === 'string') {
    focusElement = node.querySelector(firstFocusSelector);
}

if (firstFocusSelector && firstFocusSelector.current) {
    focusElement = firstFocusSelector.current;
}
```

Comment (neretin-trike) [reply packages/plasma-hope/src/hooks/useFocusTrap.ts:19]: и можешь пожалуйста изменение этого хука в отдельный коммит вынести? `feat(plasma-hope):...`

File: packages/plasma-hope/src/components/Modal/ModalView.tsx (hunk 5 above)
Comment (kayman233) [reply packages/plasma-hope/src/components/Modal/ModalView.tsx:24]: на showCloseIcon есть тест, как раз на него новые снапшоты и добавлены

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-629/

File: packages/plasma-hope/src/hooks/useFocusTrap.ts (hunk 6 above)
Comment (kayman233) [reply packages/plasma-hope/src/hooks/useFocusTrap.ts:19]: мне кажется, что текущий ифак нормально выглядит. А также в твоей варианте typescript ругается на `current`, а если это исправлять, то еще более громоздко выглядит, как мне кажется

File: packages/plasma-hope/src/components/Modal/ModalView.tsx (hunk 5 above)
Comment (neretin-trike) [reply packages/plasma-hope/src/components/Modal/ModalView.tsx:24]: а, сорри 👌, проглядел

**Comment by Salute-Eva** [comment]
Documentation preview deployed!

  website: http://plasma.sberdevices.ru/pr/pr-629/
b2c storybook: http://plasma.sberdevices.ru/pr/pr-629/b2c-storybook/
web storybook: http://plasma.sberdevices.ru/pr/pr-629/web-storybook/

File: packages/plasma-hope/src/hooks/useFocusTrap.ts (hunk 6 above)
Comment (neretin-trike) [reply packages/plasma-hope/src/hooks/useFocusTrap.ts:19]: oke 👍

**Review by neretin-trike — APPROVED** [review]

File: packages/plasma-hope/api/plasma-hope.api.md [hunk 7]
```diff
@@ -906,7 +907,13 @@ export const Modal: FC<ModalProps>;
 
 // @public (undocumented)
 export interface ModalProps extends ModalViewProps {
+    focusLaterRef?: React_2.RefObject<HTMLElement>;
+    initialFocusRef?: React_2.RefObject<HTMLElement>;
+    isCloseOnEsc?: boolean;
+    isCloseOnOverlayClick?: boolean;
```
[the remark this answers is not in the document]
Comment (kayman233) [reply packages/plasma-hope/api/plasma-hope.api.md:913]: ну кстати да, в данном случае наверное без is лучше

File: packages/plasma-hope/api/plasma-hope.api.md [hunk 8]
```diff
@@ -923,6 +930,7 @@ export interface ModalViewProps extends React_2.HTMLAttributes<HTMLDivElement> {
     children?: React_2.ReactNode;
     closeButtonAriaLabel?: string;
     onClose?: () => void;
+    showCloseIcon?: boolean;
```
[the remark this answers is not in the document]
Comment (kayman233) [reply packages/plasma-hope/api/plasma-hope.api.md]: сделал showCloseButton, так кажется получше

File: packages/plasma-hope/src/utils/focusManager.ts [hunk 9]
```diff
@@ -25,8 +25,10 @@ export class FocusManager {
     };
 
     // добавление на фокус после анмаунта
-    public markForFocusLater = () => {
-        this.focusLaterElements.push(document.activeElement as HTMLElement);
+    public markForFocusLater = (focusLaterNode?: React.RefObject<HTMLElement>) => {
```
[the remark this answers is not in the document]
Comment (kayman233) [reply packages/plasma-hope/src/utils/focusManager.ts:28]: да, ну логику focus-trap расширили

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-629/

File: packages/plasma-hope/api/plasma-hope.api.md [hunk 10]
```diff
@@ -906,7 +907,13 @@ export const Modal: FC<ModalProps>;
 
 // @public (undocumented)
 export interface ModalProps extends ModalViewProps {
+    focusLaterRef?: React_2.RefObject<HTMLElement>;
```
[the remark this answers is not in the document]
Comment (kayman233) [reply packages/plasma-hope/api/plasma-hope.api.md:912]: ну у меня в голове были еще focusAfter или focusAfterClose

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!

  website: http://plasma.sberdevices.ru/pr/pr-629/
b2c storybook: http://plasma.sberdevices.ru/pr/pr-629/b2c-storybook/
web storybook: http://plasma.sberdevices.ru/pr/pr-629/web-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Review by Yeti-or — APPROVED** [review]

**Comment by Salute-Eva** [comment]
🚀 This PR is included in version: @salutejs/plasma-b2c@1.220.0, @salutejs/plasma-core@1.125.0, @salutejs/plasma-cy-utils@0.65.0, @salutejs/plasma-hope@1.220.0, @salutejs/plasma-icons@1.154.0, @salutejs/plasma-sb-utils@0.123.0, @salutejs/plasma-temple-docs@0.160.0, @salutejs/plasma-temple@1.169.0, @salutejs/plasma-tokens-b2b@1.24.0, @salutejs/plasma-tokens-b2c@0.33.0, @salutejs/plasma-tokens-native@1.9.0, @salutejs/plasma-tokens-utils@0.29.0, @salutejs/plasma-tokens-web@1.39.0, @salutejs/plasma-tokens@1.58.0, @salutejs/plasma-typo@0.36.0, @salutejs/plasma-ui-docs@0.209.0, @salutejs/plasma-ui@1.201.0, @salutejs/plasma-web-docs@0.178.0, @salutejs/plasma-web@1.220.0, @salutejs/plasma-website@0.186.0 🚀

**Pull request merged**